File: app/src/main/java/com/apanda/base/Module/Gank/contract/ItemContract.java
package com.apanda.base.Module.Gank.contract;


import com.apanda.base.Module.Gank.bean.ItemsBean;
import com.apanda.base.base.BaseModel;
import com.apanda.base.base.BasePresenter;
import com.apanda.base.base.BaseView;

import rx.Observable;

public class ItemContract {

    public interface Model extends BaseModel {
        Observable<ItemsBean> loadCatoBean(String _type);
    }

    public interface View extends BaseView {
        void returnItemsBean(ItemsBean _gankBean);
    }

    public abstract static class Presenter extends BasePresenter<ItemContract.View, ItemContract.Model> {
    }

}

File: app/src/main/java/com/apanda/base/Module/Gank/contract/ItemContract.kt
package com.apanda.base.Module.Gank.contract


import com.apanda.base.Module.Gank.bean.ItemsBean
import com.apanda.base.base.BaseModel
import com.apanda.base.base.BasePresenter
import com.apanda.base.base.BaseView

import rx.Observable

class ItemContract {

    interface Model : BaseModel {
        fun loadCatoBean(_type: String): Observable<ItemsBean>
    }

    interface View : BaseView {
        fun returnItemsBean(_gankBean: ItemsBean)
    }

    abstract class Presenter : BasePresenter<ItemContract.View, ItemContract.Model>()

}
